### ISSUE TITLE
Fractional stat values

### ISSUE THREAD

**Issue by dorkster** [flareteam/flare-engine#1784]
See discussion here: https://opengameart.org/comment/93600

**Comment by WithinAmesia** [comment]
Does this include resistances / (arbitrary resistances?). I plan to use / create spell talents where it uses +0.5% and -0.5% arbitrary resistances. So that I could have a spell talent work to give +3% to one resistance, +1% for two other resistances and -0.5% to two other resistances and -1.5% resistance to another resistance for every single talent point invested (up to to say 30 talent points maximum). Although this relies on having +0.5% and -0.5% arbitrary resistances to squeeze more depth and progression out of the Flare game engine.

**Comment by dorkster** [comment]
The idea is to include all of the stats on the character sheet. So yes, resistances would be included.

**Comment by WithinAmnesia** [comment]
My brain is playing these together right now and I am just saying to myself 'yeah okay brain that will definitely work as a proper congratulatory reply': https://www.youtube.com/watch?v=3WAOxKOmR90&ab_channel=RescoeZ + https://www.youtube.com/watch?v=seFHSuL0nsw&ab_channel=jazz + https://www.youtube.com/watch?v=VKMw2it8dQY&ab_channel=tofu Monkey Lizard brain wins this round XD.

**Comment by WithinAmnesia** [comment]
@dorkster Does experience https://github.com/flareteam/flare-engine/wiki/Attribute-Reference#enginesettings-number-format number format include xp bar? I got like 1000 xp needed fora level and enemies giving 1.25 xp per kill at a 12.5% experience rate of a 10 xp mob kill (10 x 0.125 = 1.25 xp) for some levels but shows up as like 1 xp and I'm not sure if the xp scaling is working or like has hidden values lol?

**Comment by dorkster** [comment]
To support large values, we store XP as an unsigned long integer. While floats can technically store large integers, there just aren't enough bits to represent a very large integer *with* a fractional part.

So when XP is rewarded, we have a separate float variable to do the math with the (presumably) smaller numbers. We then subtract the integer part of the float and add it to the primary XP variable. The fractional part persists in the float variable until it exceeds 1.0, and the process repeats.

In short, the XP math should be working correctly, but you'll only be able to see it represented as whole integers.

There *is* a slight bug here: We don't save the fractional part to the player's save file when quitting. So they technically can lose a fraction of an experience point between sessions. I never considered it much of a problem, but it could be easily fixed by writing that "temporary" float to the player's save file in addition to the unsigned long XP.

**Comment by WithinAmnesia** [comment]
Ah okay that is fine. I am glad that it is working behind the scenes and yeah a small fraction of an xp point is not that big of a deal. I kinda like the simple clean xp bar look of whole integers too.
